cmd/snap/cmd_validate.go

```diff
@@ -0,0 +1,189 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+)
+
+type cmdValidate struct {
+	clientMixin
+	Monitor bool `long:"monitor"`
+	// XXX: enforce mode is not supported yet
+	Enforce    bool `long:"enforce" hidden:"yes"`
+	Forget     bool `long:"forget"`
+	Positional struct {
+		ValidationSet string `positional-arg-name:"<validation-set>"`
+	} `positional-args:"yes"`
+	colorMixin
+}
+
+var shortValidateHelp = i18n.G("List or apply validation sets")
+var longValidateHelp = i18n.G(`
+The validate command lists or applies validations sets
+`)
+
+func init() {
+	addCommand("validate", shortValidateHelp, longValidateHelp, func() flags.Commander { return &cmdValidate{} }, colorDescs.also(map[string]string{
+		// TRANSLATORS: This should not start with a lowercase letter.
+		"monitor": i18n.G("Monitor the given validations set"),
+		// TRANSLATORS: This should not start with a lowercase letter.
+		"enforce": i18n.G("Enforce the given validation set"),
+		// TRANSLATORS: This should not start with a lowercase letter.
+		"forget": i18n.G("Forget the given validation set"),
+	}), []argDesc{{
+		// TRANSLATORS: This needs to begin with < and end with >
+		name: i18n.G("<validation-set>"),
+		// TRANSLATORS: This should not start with a lowercase letter.
+		desc: i18n.G("Validation set with an optional pinned sequence point, i.e. account/name[=seq]"),
+	}})
+}
+
+// this is reused for both account and set name of "account/name" argument
+var validName = regexp.MustCompile("^[a-z][0-9a-z]+$")
```

Comment:
should we export this from asserts maybe?

Reply:
Do you mean client/asserts.go?

Reply:
no I mean the asserts package that define the underlying assertion,  maybe even as helper functions, there's already a IsValidAccountKeyName,  we could have:

IsValidAccountId
IsValidValidationSetName

?

Reply:
I think importing this from somewhere (asserts?) makes sense. I noticed that this is also different from the asserts.validValidationSetName.

Also the name "validName" is quite generic in a big package like "main".

Reply:
Yes, I'm sorry I forgot to address this remark earlier. I realized asserts already has asserts for both account and validation set name, so added the helpers for those and two extra tests in the asserts package.